src/libraries/System.Private.CoreLib/src/System/Collections/Generic/INonRandomizedEqualityComparer.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal interface INonRandomizedEqualityComparer<in T> : IEqualityComparer<T>
+    {
+        IEqualityComparer<T> GetComparerForSerialization() => GetRandomizedComparer();
```

Comment:
Why do we have two different methods? I would expect one should be enough.

Reply:
Future-proofing, mainly. Check out the discussion over at https://github.com/dotnet/runtime/issues/4761#issuecomment-624789173.

Summary: you may have types whose default hash code calculation is non randomized (e.g., `int`, `long`). If the dictionary starts to become unbalanced, you'll want to fall back to a randomized hash code calculation routine. But the _default_ comparer is the non-randomized one, not the randomized one, so `GetComparerForSerialization` would return the non-randomized comparer.

Reply:
We should add these abstractions only once they are required.